backend/dataall/base/utils/logs_utils.py

```diff
@@ -0,0 +1,33 @@
+from typing import Optional, Callable, List, Any
+from dataall.base.config import config
+from dataall.base.utils.decorator_utls import process_func
+
+
+def is_feature_has_allowed_values(
+    allowed_values: List[Any], resolve_property: Optional[Callable] = None, config_property: Optional[str] = None
+) -> object:
+    def decorator(f):
+        fn, fn_decorator = process_func(f)
+
+        def decorated(*args, **kwargs):
+            config_property_value = None
+            if config_property is None and resolve_property is None:
+                raise Exception('Config property not provided')
+            if resolve_property:
+                config_property_value = resolve_property(*args, **kwargs)
+            value = config.get_property(config_property_value)
+            if value not in allowed_values:
+                raise Exception(
+                    f'Disabled since incorrect values in config {config_property_value}. Correct config values {allowed_values}'
+                )
+            return fn(*args, **kwargs)
+
+        return fn_decorator(decorated)
+
+    return decorator
+
+
```

Comment:
I think I was not very clear with the purpose of `is_feature_has_allowed_values`. It was not only to check the validity of the config values (although I really like that you added it). It was to enable or disable a feature depending on the value, same as is_feature_enabled but with string values instead of booleans. So it would be more a `is_feature_enabled_values`. And in this case the enabled_values=['enabled', 'admin-only']. We will implement the difference between `enabled` and `admin-only` separately

Reply:
Yes it would be more like `is_feature_enable_on_allowed_values` ( I can rename the function according to what makes sense ). As you said, this decorator is just to check if the config has correct config values since we are departing from true/false ( boolean values ) . The logic to check based on the config value can be done in the service layer. 

Initially I was trying not to touch code in the service layer and just make use of decorator to allow/disallow some feature, but this was conflicting with how the code is structured.